opm/output/eclipse/Summary.cpp

```diff
@@ -1075,6 +1122,59 @@ Summary::Summary( const EclipseState& st,
     }
 }
 
+std::vector< std::pair< std::string, double > >
+efficiency_factors( const smspec_node_type* type,
+                    const Schedule& schedule,
+                    const std::vector< const Well* >& schedule_wells,
+                    size_t timestep ) {
+    std::vector< std::pair< std::string, double > > efac;
+
+    if(    smspec_node_get_var_type(type) != ECL_SMSPEC_GROUP_VAR
+        && smspec_node_get_var_type(type) != ECL_SMSPEC_FIELD_VAR
+        && smspec_node_get_var_type(type) != ECL_SMSPEC_REGION_VAR
+        && !smspec_node_is_total( type ) ) {
+        return efac;
+    }
+
+    const bool is_group = smspec_node_get_var_type(type) == ECL_SMSPEC_GROUP_VAR;
+    const bool is_rate = !smspec_node_is_total( type );
+    const auto &groupTree = schedule.getGroupTree(timestep);
+
+    /* Preferably we would like to use a native validation method ("hasParent")
+     * instead of a try-catch block.
+     * When/if that function is implemented, change the current workflow.*/
+    auto getparent = [&]( const Group* g ) -> decltype( g ) {
+        try {
+            const auto& parent = groupTree.parent( g->name() );
+            if( !schedule.hasGroup( parent ) ) return nullptr;
+            return &schedule.getGroup( parent );
+        }
+        catch( const std::invalid_argument& ) { return nullptr; }
+    };
+
+    for( const auto* well : schedule_wells ) {
+        double eff_factor = well->getEfficiencyFactor(timestep);
+
+        if (!schedule.hasGroup(well->getGroupName(timestep)))
+            continue;
```

Comment:
Ehhh - is this `continue` correct - seems the iteration will be terminated *without* actually adding the well, in the `efac` vector?

Reply:
Currently OPM doesn't handle wells without groups (setting group to default in WELSPECS result in throwing on run-time). However, when the well is not yet defined it will not have a group name, causing schedule.getGroup to throw. There is another function well.isActive(timestep) that will do the same and is perhaps more descriptive of what is really going on here. Or perhaps line 1161-1171 should be protected by an if( schedule.hasGroup(well->getGroupName(timestep)) ) so that this code is not dependent on the well having a group when getting WEFAC.